app/controllers/coronavirus_controller.rb

```diff
@@ -0,0 +1,77 @@
+class CoronavirusController < ApplicationController
+  before_action :require_coronavirus_editor_permissions!
+  layout "admin_layout"
+
+  CONTENT_ID = "774cee22-d896-44c1-a611-e3109cce8eae".freeze
+  CONTENT_URL = "https://raw.githubusercontent.com/alphagov/govuk-coronavirus-content/master/content/coronavirus_landing_page.yml".freeze
+
+  def index; end
+
+  def update_draft
+    response = RestClient.get(CONTENT_URL)
+
+    if response.code == 200
+      corona_content = YAML.safe_load(response.body)["content"]
+
+      if valid_content?(corona_content)
+        presenter = CoronavirusPagePresenter.new(corona_content)
+
+        Services.publishing_api.put_content(CONTENT_ID, presenter.payload)
+        flash["notice"] = "Draft content updated"
+      end
+    else
+      flash["alert"] = "Error received from Github - #{response.code}"
+    end
+
+    redirect_to coronavirus_path
+  end
+
+  def publish
+    begin
+      Services.publishing_api.publish(CONTENT_ID, update_type)
+
+      flash["notice"] = "Page published!"
+    rescue GdsApi::HTTPConflict
+      flash["alert"] = "Page already published - update the draft first"
+    end
+
+    redirect_to coronavirus_path
+  end
+
+private
+
+  def update_type
+    major_update? ? "major" : "minor"
+  end
+
+  def major_update?
+    params["update-type"] == "major"
+  end
+
+  def valid_content?(content)
+    return false if content.nil?
+
+    missing_keys = (required_landing_page_keys - content.keys)
+    if missing_keys.any?
+      flash["alert"] = "Invalid content - please recheck Github and add #{missing_keys.join(', ')}."
+      return false
+    end
+
+    true
+  end
+
+  def required_landing_page_keys
```

Comment:
We should write some guidance that if a new section is added to the design and yaml we need to update this method, and the corresponding method [here](https://github.com/alphagov/collections/blob/981bfd4700851275d5ff4f166bc158fe58d3ba88/app/presenters/coronavirus_landing_page_presenter.rb#L7)

Reply:
👍

Reply:
This will continue to pass if extra keys are added, but yep it'd be good to keep updated